jacqueline-chamnankool/wk01/4-thurs/geometry.js

```diff
@@ -0,0 +1,95 @@
+//rectangle
+const rectangleA = {
+    length: 4,
+    width: 4
+};
+
+function isSquare(rectangle) {
+    if (rectangle.length === rectangle.width) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+function area(rectangle) {
+    return rectangle.length * rectangle.width;
+}
+
+function perimeter(rectangle) {
+    return (2 * rectangle.length) + (2 * rectangle.width);
+}
+
+console.log("-".repeat(50));
+console.log(isSquare(rectangleA));
+console.log(area(rectangleA));
+console.log(perimeter(rectangleA));
+
+//triangle
+const triangleA = {
+    sideA: 3,
+    sideB: 4,
+    sideC: 4
+};
+const triangleB = {
+    sideA: 3,
+    sideB: 3,
+    sideC: 3
+};
+const triangleC = {
+    sideA: 3,
+    sideB: 2,
+    sideC: 4
+};
+
+function isEquilateral(triangle) {
+    if (triangle.sideA === triangle.sideB && triangle.sideB === triangle.sideC) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+function isIsosceles(triangle) {
+    //two sides have the same length
+    if (triangle.sideA === triangle.sideB || triangle.sideA === triangle.sideC) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+function areaTriangle(triangle) {
+    //holy shit
+    let s = (triangle.sideA + triangle.sideB + triangle.sideC) / 2;
+    return Math.sqrt(s*((s*triangle.sideA)*(s*triangle.sideB)*(s*triangle.sideC))).toFixed(2);
```

Comment:
Style-wise this could be made more readable but by your comment, I'd bet you're already aware of that.